packages/governance/src/binaryBallotCounter.js

```diff
@@ -23,60 +30,71 @@ const makeQuorumCounter = quorumThreshold => {
   return Far('checker', { check });
 };
 
-// Exported for testing purposes
-const makeBinaryBallotCounter = (
-  question,
-  positions,
-  threshold,
-  tieOutcome = undefined,
-  closingRule,
-  instance,
-) => {
+const validateBinaryBallotSpec = ballotSpec => {
+  looksLikeBallotSpec(ballotSpec);
+
+  const positions = ballotSpec.positions;
   assert(
     positions.length === 2,
     X`Binary ballots must have exactly two positions. had ${positions.length}: ${positions}`,
   );
-  assert.typeof(question, 'string');
-  assert.typeof(positions[0], 'string');
-  assert.typeof(positions[1], 'string');
-  if (tieOutcome) {
-    assert(
-      positions.includes(tieOutcome),
-      X`The default outcome on a tie must be one of the positions, not ${tieOutcome}`,
-    );
-  }
-
-  const template = buildBallot(
-    ChoiceMethod.CHOOSE_N,
-    question,
-    positions,
-    1,
-    instance,
+  assert(
+    ballotSpec.maxChoices <= positions.length,
+    X`Choices must not exceed length`,
   );
-  const ballotDetails = template.getDetails();
 
+  looksLikeQuestionForType(ballotSpec.electionType, ballotSpec.question);
+
+  assert(
+    ballotSpec.maxChoices === 1,
+    X`Can only choose 1 item on a binary ballot`,
+  );
   assert(
-    ballotDetails.method === ChoiceMethod.CHOOSE_N,
-    X`Binary ballot counter only works with CHOOSE_N`,
+    ballotSpec.method === ChoiceMethod.CHOOSE_N,
+    X`${ballotSpec.method} must be CHOOSE_N`,
   );
+  // We don't check the quorumRule or quorumThreshold here. The quorumThreshold
+  // is provided by the Registrar that creates this ballotCounter, since only it
+  // can translate the quorumRule to a required number of votes.
+};
+
+// Notice that BinaryBallotCounter is designed to run as a Zoe contract. The
+// business part of the contract is extracted here so it can be tested
+// independently. The standard Zoe start function is at the bottom of this file.
+
+/** @type {BuildBallotCounter} */
+const makeBinaryBallotCounter = (ballotSpec, threshold, instance) => {
+  validateBinaryBallotSpec(ballotSpec);
+
+  const ballot = buildBallot(ballotSpec, instance);
+  const details = ballot.getDetails();
+  // @ts-ignore typescript is confused here. Dunno why. The declarations clearly
+  // say Ballot.getDetails() returns a ballotDetails, which has a handle.
+  const { handle } = details;
+
   let isOpen = true;
+  const positions = ballotSpec.positions;
   const outcomePromise = makePromiseKit();
   const tallyPromise = makePromiseKit();
+  // The Registrar is responsible for creating a unique seat for each voter.
+  // This ballotCounter allows voters to re-vote, and replaces their previous
+  // choice with the new selection.
   const allBallots = makeStore('seat');
 
-  const recordBallot = (seat, filledBallotP, shares = 1n) => {
+  const submitVote = (seat, filledBallotP, shares = 1n) => {
```

Comment:
```suggestion
  /** @type {SubmitVote} */
  const submitVote = (seat, filledBallotP, shares = 1n) => {
```
Without this, TS couldn't figure out what `submitVote`'s type.

Reply:
done